cli_client/python/timesketch_cli_client/cli.py

```diff
@@ -0,0 +1,146 @@
+from .version import get_version
+
+
+class TimesketchCli(object):
+    """Timesketch CLI state object.
+
+    Attributes:
+        sketch_from_flag: Sketch ID if provided by flag
+        config_assistant: Instance of ConfigAssistant
+    """
+    def __init__(self, api_client=None, sketch_from_flag=None, conf_file=None):
+        """Initialize the state object.
+
+        Args:
+            sketch_from_flag: Sketch ID if provided by flag.
+        """
+        self.api = api_client
+        self.sketch_from_flag = sketch_from_flag
+
+        if not api_client:
+            try:
+                self.api = timesketch_config.get_client(load_cli_config=True)
+                if not self.api:
+                    raise RequestConnectionError
+            except RequestConnectionError:
+                click.echo('ERROR: Cannot connect to the Timesketch server.')
+                sys.exit(1)
+
+        self.config_assistant = timesketch_config.ConfigAssistant()
+        if conf_file:
+            self.config_assistant.load_config_file(
+                conf_file, load_cli_config=True)
+        else:
+            self.config_assistant.load_config_file(load_cli_config=True)
+
+    @property
+    def sketch(self):
+        """Sketch object from the API client.
+
+        Returns:
+            Sketch object.
+        """
+        active_sketch = None
+        sketch_from_config = self.config_assistant.get_config('sketch')
+
+        if self.sketch_from_flag:
+            active_sketch = self.api.get_sketch(
+                sketch_id=int(self.sketch_from_flag))
+        elif sketch_from_config:
+            active_sketch = self.api.get_sketch(
+                sketch_id=int(sketch_from_config))
+
+        if not active_sketch:
+            click.echo(('ERROR: You need to specify a sketch, either with a '
+                        'flag (--sketch <SKETCH ID>) or update the config.'))
+            sys.exit(1)
+
+        # Make sure we have access to the sketch.
+        try:
+            active_sketch.name
+        except KeyError:
+            click.echo('ERROR: No such sketch or you don\'t have access.')
+            sys.exit(1)
+
+        return active_sketch
+
+    @property
+    def output_format(self):
+        """Get or set the default output format.
+
+        Returns:
+            Output format as a string.
+        """
+        output_format = self.config_assistant.get_config('output_format')
+        if not output_format:
+            self.config_assistant.set_config('output', DEFAULT_OUTPUT_FORMAT)
+            self.config_assistant.save_config()
+            output_format = DEFAULT_OUTPUT_FORMAT
+        return output_format
+
+
+@click.group(context_settings={'help_option_names': ['-h', '--help']})
+@click.version_option(version=get_version(), prog_name='Timesketch CLI')
+@click.option('--sketch', type=int, default=None, help='Sketch to work in.')
+@click.pass_context
+def cli(ctx, sketch):
+    """
+    Timesketch CLI client.
```

Comment:
nit, the first line in the same place as `""""`

Reply:
Done